src/api/goods.ts

```diff
@@ -6,6 +6,19 @@ export function getAll(): Promise<Good[]> {
     .then(response => response.json());
 }
 
-export const get5First = () => {};
+export const getFirst5Goods = (): Promise<Good[]> => {
+  return getAll()
+    .then((goods: Good[]) => (
+      goods
+        .sort((curr: Good, next: Good) => curr.name.localeCompare(next.name))
+        .slice(0, 5)
+    ));
+};
 
-export const getRedGoods = () => {};
+export const getRedGoods = (): Promise<Good[]> => {
+  return getAll()
+    .then((goods: Good[]) => (
+      goods
+        .filter((good: Good) => good.color === 'red')
```

Comment:
If you have a function where the return value type is specified (i.e. `Promise<Good[]>`), you can avoid explicitly indicating types further (i.e. `goods: Good[]`, `good: Good`). However, that's a matter of taste